### PR TITLE
Properly log panics with slog

### PR DESCRIPTION
**Pull request by jxs**
## Questions
- Feel free to suggest a better location for the `panic::set_hook` as I am not 100% sure about it.

example of logs with this PR:
```
Jan 16 20:58:59.698 INFO Updated local ENR                       udp6: None, tcp6: None, tcp4: Some(9000), udp4: None, ip4: 
None, id: 0x43c7..6f12, seq: 245, enr: enr:-La4QHvi7qEh13FeUFdp-c-4Uokn17MoHAEqFxmLKpKaWjRBFWWNKHDYw2zLMSWZlkz0GGdP0D3tsUIKYX_E0R2C9ziB9YdhdHRuZXRziAAAAAAAYAAAhGV0aDKQYolB7wQAECAAiQMAAAAAAIJpZIJ2NIRxdWljgiMpiXNlY3AyNTZrMaEDShdD_qHrRAQ0Ogm2JuIneubXoaYuvFt4X8d_eFAeTkiIc3luY25ldHMAg3RjcIIjKA, service: libp2p
Jan 16 20:58:59.699 INFO Updated local ENR                       udp6: None, tcp6: None, tcp4: Some(9000), udp4: None, ip4: None, id: 0x43c7..6f12, seq: 246, enr: enr:-La4QJ3XfEKbI7K7FMjynvr0CwNSy6hYxZ1KystbfbKp4fC_bju5Xg-rOVJOJV_viQBaagBzvKII1SlsUfIl4_n8XniB9odhdHRuZXRziAAAAAAAYAAAhGV0aDKQYolB7wQAECAAiQMAAAAAAIJpZIJ2NIRxdWljgiMpiXNlY3AyNTZrMaEDShdD_qHrRAQ0Ogm2JuIneubXoaYuvFt4X8d_eFAeTkiIc3luY25ldHMAg3RjcIIjKA, service: libp2p
Jan 16 20:59:00.028 CRIT Task panic. This is a bug!              message: Panic in the streets of London!, location: /home/jxs/dev/oss/rust-libp2p/protocols/gossipsub/src/behaviour.rs:3344:19
Jan 16 20:59:00.038 INFO Saved DHT state                         service: network
Jan 16 20:59:00.039 INFO Network service shutdown                service: network
Jan 16 20:59:00.039 INFO Internal shutdown received              reason: Panic (fatal error)
Jan 16 20:59:00.040 INFO Shutting down..                         reason: Failure("Panic (fatal error)")
Jan 16 20:59:00.263 INFO Saved beacon chain to disk              service: beacon
Jan 16 20:59:00.733 INFO Execution payloads are pruned           service: freezer_db
Jan 16 20:59:09.819 INFO UPnP not available                      error: IO error: Resource temporarily unavailable (os error 11), service: UPnP
Panic (fatal error)
```

thanks

### CODE REVIEW

File: beacon_node/src/lib.rs [hunk 1]
```diff
@@ -67,6 +68,20 @@ impl<E: EthSpec> ProductionBeaconNode<E> {
         let blobs_db_path = client_config.create_blobs_db_path()?;
         let executor = context.executor.clone();
 
+        // Log panics properly.
+        {
+            let log = log.clone();
+            std::panic::set_hook(Box::new(move |info| {
+                crit!(
+                    log,
+                    "Task panic. This is a bug!";
```
Comment (dapplion) [inline beacon_node/src/lib.rs]: Should it suggest to contact developers / open a Github issue if this happens?

At least like in the previous version
```
                            "advice" => "Please check above for a backtrace and notify \
                                         the developers"
```

Comment (jxs) [reply beacon_node/src/lib.rs]: yeah makes sense to keep it, thanks!

**Review by AgeManning — APPROVED** [review]
LGTM!

**Comment by AgeManning** [comment]
@mergifyio queue

**Pull request merged**